### PR TITLE
(RE-13380) Support Apple Notarization for Puppet Packages.

### PR DESCRIPTION
**Pull request by jackie-kinsler**
Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

File: lib/vanagon/platform/osx.rb [hunk 1]
```diff
@@ -56,8 +58,26 @@ def generate_package(project) # rubocop:disable Metrics/AbcSize
...
+         #  The signing commands below should not cause `vanagon build` to fail. Many devs need to run `vanagon build`
+         #  locally and do not necessarily need signed packages. The `|| :` will rescue failures by evaluating as successful.
+         #  /Users/binaries will be a list of all binaries that need to be signed
+         "touch /Users/binaries || :",
+         #  Find all of the executables (Mach-O files), and put the in /Users/binaries
+         "for item in `find $(tempdir)/osx/build/ -perm -0100 -type f` ; do file $$item | grep 'Mach-O' ; done | awk '{print $$1}' | sed 's/\:$$//' > /Users/binaries || :",
+         #  A tmpdir is created on the signing_host, all of the executables will be rsyncd there to be signed
+         "#{Vanagon::Utilities.ssh_command}  #{signing_host} mkdir -p /tmp/$(binaries_dir) || :",
+         "rsync -e '#{Vanagon::Utilities.ssh_command}' --no-perms --no-owner --no-group --files-from=/Users/binaries / #{signing_host}:/tmp/$(binaries_dir) || :",
+         "rsync -e '#{Vanagon::Utilities.ssh_command}' --no-perms --no-owner --no-group /Users/binaries #{signing_host}:/tmp/binaries_list || :",
```
Comment (e-gris) [inline lib/vanagon/platform/osx.rb:70]: Something here doesn't seem consistent: `/tmp/$(binaries_dir)` vs. `/tmp/binaries_list`

**Review by e-gris — COMMENTED** [review]
Questions, mostly.

File: lib/vanagon/platform/osx.rb (hunk 1 above)
Comment (jackie-kinsler) [reply lib/vanagon/platform/osx.rb:70]: /tmp/$(binaries_dir) contains all of the binaries that are being rsynced
/tmp/binaries_list is a simple file that contains the list of binaries, which is then submitted to the signing script for ease of use

File: lib/vanagon/platform/osx.rb [hunk 2]
```diff
@@ -56,8 +58,26 @@ def generate_package(project) # rubocop:disable Metrics/AbcSize
...
          bom_install,
 
+         #  The signing commands below should not cause `vanagon build` to fail. Many devs need to run `vanagon build`
+         #  locally and do not necessarily need signed packages. The `|| :` will rescue failures by evaluating as successful.
+         #  /Users/binaries will be a list of all binaries that need to be signed
+         "touch /Users/binaries || :",
+         #  Find all of the executables (Mach-O files), and put the in /Users/binaries
+         "for item in `find $(tempdir)/osx/build/ -perm -0100 -type f` ; do file $$item | grep 'Mach-O' ; done | awk '{print $$1}' | sed 's/\:$$//' > /Users/binaries || :",
+         #  A tmpdir is created on the signing_host, all of the executables will be rsyncd there to be signed
+         "#{Vanagon::Utilities.ssh_command}  #{signing_host} mkdir -p /tmp/$(binaries_dir) || :",
```
[the remark this answers is not in the document]
Comment (jackie-kinsler) [reply lib/vanagon/platform/osx.rb:68]: binaries_dir is created in the Makefile.erb

File: lib/vanagon/platform/osx.rb (hunk 1 above)
Comment (jackie-kinsler) [reply lib/vanagon/platform/osx.rb:70]: I like having the /tmp/binaries_list file as opposed to just operating on the directory, as things are buried in two levels of temp dirs... and searching through the build logs is annoying vs simply doing `cat /tmp/binaries_list`

**Pull request merged**